### PR TITLE
Added webdriver.io tests for base

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
Added webdriver.io tests for terra-base

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const webpackPort = 8080;
 // Flex specs search between local pacakge and repo
 let specs = path.join('tests', 'wdio', '**', '*-spec.js');
 if (__dirname === process.cwd()) {
-  specs = path.join('packages', '**', specs);
+  specs = path.join('packages', '*', specs);
```
Comment (mhemesath) [inline wdio.conf.js]: so this fixed a nasty issue I discovered where a double glob was causing wdio to pull in tests from the node_modules directory of every package. This means that when I pushed base tests, every component that used base reran those tests. Switching to a single glob excludes those node_module directories.

File: packages/terra-base/tests/nightwatch/SwitchLocaleBase.jsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ class SwitchLocaleBase extends React.Component {
         locale={this.state.locale}
         customMessages={customMessages[this.state.locale]}
       >
-        <FormattedMessage id="App.locale" />
+        <div id="message"><FormattedMessage id="App.locale" /></div>
```
Comment (mhemesath) [inline packages/terra-base/tests/nightwatch/SwitchLocaleBase.jsx]: added an id to make test more predictable on div fetch

File: packages/terra-base/tests/nightwatch/ArrayBase.jsx [hunk 3]
```diff
@@ -5,14 +5,10 @@ const ArrayBase = () => (
   <div>
     <Base>
       <div id="one">One</div>
```
Comment (mhemesath) [inline packages/terra-base/tests/nightwatch/ArrayBase.jsx]: only one child is needed to verify this works

File: wdio.conf.js (hunk 1 above)
Comment (mhemesath) [reply wdio.conf.js]: @emilyrohrbough if any other projects integrated webdriver.io they may want to update this

Comment (emilyrohrbough) [reply wdio.conf.js]: @mhemesath I'll work on getting the clinical PR updated

File: packages/terra-base/tests/nightwatch/ArrayBase.jsx (hunk 3 above)
Comment (bjankord) [reply packages/terra-base/tests/nightwatch/ArrayBase.jsx]: This test was to show multiple sibling children can be passed to Base, we'll want to keep the multiple sibling children.

Comment (mhemesath) [reply packages/terra-base/tests/nightwatch/ArrayBase.jsx]: Technically this is a better test for jest IMO

File: packages/terra-base/tests/nightwatch/base-spec.js [hunk 4]
```diff
@@ -1,43 +0,0 @@
...
-  'Switches locale and displays en-US customized locale message': (browser) => {
-    browser
-      .url(`${browser.launchUrl}/#/tests/base-tests/switch`)
-      .waitForElementVisible('button', waitInms)
-      .click('button');
-
-    browser.expect.element('div').text.to.contain('en-US').before(waitInms);
-  },
-
-  // NOTE: This test is designed to assume that tests won't fail on an error.  If it starts failing,
```
Comment (emilyrohrbough) [inline packages/terra-base/tests/nightwatch/base-spec.js]: Should we move this comment forward to the wdio spec?

**Pull request merged**